reana_server/deleter.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2023 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""REANA utilities to delete file in a workspace."""
+
+import errno
+from enum import Enum
+from pathlib import Path
+from typing import Union
+
+import click
+from reana_commons import workspace
+from reana_db.models import Workflow
+
+from reana_server.utils import is_relative_to
+
+
+class InOrOut(Enum):
+    """Enumeration of the possible combinations of input / output files that can be preserved by the deleter."""
+
+    INPUTS = 1
+    OUTPUTS = 2
+    INPUTS_OUTPUTS = 3
+    NONE = 4
+
+
+class Deleter:
```

Comment:
Could you please write a few tests to check that `Deleter` works as expected when using different modes (`NONE`, `INPUTS_OUTPUTS`, ...)?